src/models/curriculum/unit.ts

```diff
@@ -194,17 +193,15 @@ const populateProblemSections = async (content: Record<string, any>, unitUrl: st
 
 export function getUnitJson(unitId: string | undefined, appConfig: AppConfigModelType) {
   const unitSpec = getUnitSpec(unitId, appConfig);
-  const unitPath = unitSpec?.content;
-  const unitUrl = getAssetUrl(unitPath!);
+  const unitUrl = unitSpec?.content;
   return fetch(unitUrl!)
            .then(async response => {
              if (response.ok) {
                const unitContent = await response.json();
-               const fullUnitContent = unitContent && populateProblemSections(unitContent, unitUrl);
+               const fullUnitContent = unitContent && populateProblemSections(unitContent, unitUrl!);
```

Comment:
As I understand it, `getUnitSpec` will always return data for the specified unit or the default unit if either a) no unit is specified, or b) no matching unit is found, _and_ `appConfig.defaultUnit` is defined. In practice `appConfig.defaultUnit` is always defined, though it's technically possible for it to not be defined.

I think this is one of those areas where we're not doing a good job of handling bad unit codes. Instead of showing an error, we're just showing the default unit. I'm not sure that needs to be fixed as part of this work, but I can look into it. Maybe it'd be enough to show the error pop-up/dialog with an error saying something like "Unit not found. Please check the value of the unit specified in the URL..."?

Reply:
I agree that handling errors by defaulting to Stretching and Shrinking should not be considered part of this work, but it _has_ been an ongoing source of confusion and there should be a story about coming up with a better solution.

Reply:
I made a new story for this issue: https://www.pivotaltracker.com/story/show/184753375